File: docs/test-beds/javascript/fixture.js
// JavaScript test bed
import { Helper } from './helper.js';

export class Container {
  constructor() { this.items = []; }
  add(item) { this.items.push(item); }
  size() { return this.items.length; }
}

export function process(input) {
  const c = new Container();
  c.add(input);
  Helper.log(c);
  return c;
}

const FOO = 42;

File: docs/test-beds/ruby/fixture.rb
# Ruby test bed
require_relative 'helper'

class Box
  attr_accessor :value
  def initialize(value); @value = value; end
end

class Container
  def initialize; @items = []; end
  def add(item); @items << item; end
  def size; @items.length; end
end

def process(input)
  c = Container.new
  c.add(input)
  Helper.log(c)
  c
end

FOO = 42
